docs-source/content/userguide/cicd/choose-an-approach.md

```diff
@@ -0,0 +1,59 @@
+---
+title: "Choose an Approach"
+date: 2019-04-11T13:36:57-04:00
+draft: false
+weight: 3
+decsription: "How to choose an approach."
+---
+
+Let's review what we have discussed and talk about when we might want to use 
+various approaches.  We can start by asking ourselves questions like these:
+
+
+- *Can you make the desired change with a configuration override?*  
+  The WebLogic Kubernetes Operator allows you to inject a number of [configuration 
+  overrides]({{< relref "/userguide/managing-domains/configoverrides/_index.md" >}}) 
+  into your pods at startup time.  This allows you to use the same image for multiple 
+  different configurations.  A good example would be changing the settings for a data 
+  source, for example. You may wish to have a larger connection pool in your production 
+  environment than you do in your development/test environments.  You probably also 
+  want to have different credentials.  You may want to change the service name, and 
+  so on.  All of these kinds of updates can be made with configuration overrides.
+  These are placed in a Kubernetes config map, i.e. they are outside the image, so 
+  they do not require rebuilding the Docker image.  If all of your changes fit into 
+  this category, it is probably much better to just use the configuration overrides 
+  instead of building a new image.  
+- *Are you only changing the WebLogic configuration, i.e. deploying or updating an 
+  application, changing a resource configuration in a way that is not supported by 
+  configuration overrides, etc.?*  
+  If your changes fit into this category, and you have used the "domain-in-image" 
+  approach and the Docker layering model, then you only need to update the top layer 
+  of your image.  This is relatively easy compared to making changes in lower layers.
+  You could create a new layer with the changes, or your could rebuild/replace the 
+  existing top layer with a new one.  Which approach you choose depends most on 
+  whether you need to maintain the same domain encryption keys or not.
```

Comment:
Maybe add something like:    There's also trade-off in coming up with edits to the current configuration via a custom additive WDT/off-line script, rather than rerunning an changed version of the original WDT/off-line script.   The latter tends to be easier, but see the next bullet item before you consider it.